src/MockBot.Api/Controllers/ChatController.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using MockBot.Api.Interfaces;
+using MockBot.Api.Models;
+
+namespace MockBot.Api.Controllers
+{
+    [Route("api/chats")]
+    [ApiController]
+    public class ChatController : ControllerBase
+    {
+        private readonly IChatService _chatService;
+
+        public ChatController(IChatService chatService)
+        {
+            _chatService = chatService;
+        }
+
+        [HttpGet]
+        public OkObjectResult FindAll()
+        {
+            var chats = _chatService.FindAll();
+
+            return Ok(chats);
+        }
+
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Chat))]
```

Comment:
The controller looks good, just as a minor point, with ActionResult return type you can specify the return type in the ActionResult object, which will allow you to remove the Type = typeof(T) part and also return T since T directly returns as new ObjectResult(T)
```suggestion
        [ProducesResponseType(StatusCodes.Status200OK)]
```

Reply:
Thanks, didn't know that!